tests/TestCase/TestSuite/IntegrationTestTraitTest.php

```diff
@@ -1356,6 +1356,26 @@ public function testSendFile(): void
         $this->assertFileResponse(TEST_APP . 'TestApp' . DS . 'Controller' . DS . 'PostsController.php');
     }
 
+    /**
+     * Test sending file in requests.
+     */
+    public function testSendUnlinked(): void
+    {
+        $filename = TMP . 'cake_test_send_unlink.php';
+        file_put_contents(
+            $filename,
+            file_get_contents(
+                TEST_APP . 'TestApp' . DS . 'Controller' . DS . 'PostsController.php'
+            )
+        );
+
+        $this->get('/posts/file?file=' . $filename);
```

Comment:
Probably this line is not right. DS would be encoded for web request, so the assertFileResponse compared the default `__FILE__` value, so that the file pointer wouldn't grab the file which you want. Then this code can find the file and remove it.

It is better to change the filename not including the DS.